### ISSUE TITLE
Javanica echo test failure

### ISSUE THREAD

**Issue by mattrjacobs** [Netflix/Hystrix#733]
Reported in Google Group: https://groups.google.com/forum/#!topic/hystrixoss/0zw3ffAB3g4

/cc @dmgcodevil 

I am getting a failure on the Javanica echo test, whihc should return an Integer but returns a HystrixExecutableBase instead.

I noticed that Execution type for the test reurns ASYNCHRONOUS, and I am worndering if this is a java version issue?

    @HystrixCommand
    public <T> T echo(T value) {
        return value;
    }

basically type <T> type.isAssignableFrom(Future.class) is true

I am about to extend the annotation for some cusotm code and want to make sure I get the tests setup correct first.

Thoughts?

**Comment by dmgcodevil** [comment]
Hi @mattrjacobs,

I see that we have a test to check javanica processing generic methods and I added new one:

Tests
```java
@Test
    public void should_work_with_parameterized_method() throws Exception {
        assertEquals(Integer.valueOf(1), userService.echo(1));

        assertEquals(1, HystrixRequestLog.getCurrentRequest().getAllExecutedCommands().size());
        assertTrue(getCommand().getExecutionEvents().contains(HystrixEventType.SUCCESS));
    }

    @Test
    public void should_work_with_parameterized_asyncMethod() throws Exception {
        assertEquals(Integer.valueOf(1), userService.echoAsync(1).get());

        assertEquals(1, HystrixRequestLog.getCurrentRequest().getAllExecutedCommands().size());
        assertTrue(getCommand().getExecutionEvents().contains(HystrixEventType.SUCCESS));
    }
```
Commands:
```java
@HystrixCommand
        public <T> T echo(T value) {
            return value;
        }

        @HystrixCommand
        public <T> Future<T> echoAsync(final T value) {
            return new AsyncResult<T>() {
                @Override
                public T invoke() {
                    return value;
                }
            };
        }
```
All tests are completed successfully. I didn't notice HystrixExecutableBase as return type. What java version you're using? Could you provide a test to reproduce the issue?.

**Comment by mattrjacobs** [comment]
This was just a crosspost from the HystrixOSS Google Group.  I've asked the OP there to chime in here.  Thanks for jumping in @dmgcodevil

**Comment by mattrjacobs** [comment]
Closing due to inactivity.  Please re-open if needed
